app/components/edit/EditSidebar.jsx

```diff
@@ -69,6 +69,12 @@ const EditSidebar = ({
       </button>,
     );
   }
+  // populate existing services so they show up on the sidebar
+  const allServices = newServices;
+  for (let i = 0; i < resource.services.length; i++) {
+    const resourceService = resource.services[i];
+    allServices[resourceService.id].name = resourceService.name;
```

Comment:
What you're doing here isn't safe, even if it happens to work now. You're mutating the values of an object that is being passed into this function (`newServices`), which causes the values to change from underneath the caller. This could cause unintentional results if the caller assumed that the values passed in as arguments would not change. You really should be making a copy of each object and modifying that copy rather than mutating the array that you passed in.

On a related note, the C-style for loop you've written is unidiomatic JavaScript. Instead of creating a loop counter, defining a loop invariant, and incrementing the loop counter, it's more encouraged these days to use the functions that allow you to directly iterate over the elements of an array.

You can address both of these points I've mentioned by using the functional programming methods `.reduce()` and `.forEach()`. This would look like the following:

```js
// Do a 2-level-deep clone of the newServices object
const allServices = newServices.entries().reduce(
  (acc, [id, service]) => ({...acc, [id]: {...service}}),
  {},
);
resource.services.forEach(service => {
  allServices[service.id].name = service.name;
});
```

Reply:
@richardxia thanks for the notes! javascript is new territory for me :p